src/index.js

```diff
@@ -6,20 +6,36 @@
  * @param {number} startFrom - index of the candidate to start further exploration from.
  * @return {number[][]}
  */
+const combinationSumRecursive = (candidates, remainingSum, finalCombinations, currentCombination, startFrom) => {
+  if (remainingSum === 0) {
+    finalCombinations.push(currentCombination.slice());
+    currentCombination = [];
+    return finalCombinations;
+  }  
+  else {
+    if (remainingSum < 0)
+      return [];
+    else {
+      for (let currentIndex = startFrom; currentIndex < candidates.length; currentIndex++) {
+        const currentNumber = candidates[currentIndex];
+          currentCombination.push(currentNumber);
+        combinationSumRecursive(candidates, remainingSum - currentNumber, finalCombinations, currentCombination, currentIndex);
+        currentCombination.pop();
```

Comment:
Puedes evitarte el push and pop si usas descomposición al llamar la función:
```js
combinationSumRecursive(
    candidates, 
    remainingSum - currentNumber, 
    finalCombinations, 
    [ ...currentCombination, currentNumber], 
    currentIndex
);
```

Reply:
Mejorado en el siguiente commit. Gracias